### DISCUSSION TITLE
compatibility with PyLandStats

### DISCUSSION THREAD

**Discussion by mohdalattar** [pysal/momepy#449] [Ideas]
I would be a really good idea to come up with a function that bridge between OSMnx and PyLandStats. Image I extract urban green spaces from OSMnx say

 gdf = ox.geometries.geometries_from_place('Manhattan', tags={'leisure': 'park', 'landuse': 'grass'})

Meaning basic landscape metrics will be extremely beneficial to calculate patch indices for green spaces which is a common practice using imagery via Fragstats. I believe this would be a great extension for "Measuring Diversity" part [http://docs.momepy.org/en/stable/user_guide/weights/diversity.html]

**Comment by martinfleis** [comment]
As far as I remember, `pylandstats` is a raster-oriented library. `momepy` is a vector one. And the bridge between a GeoDataFrame coming from the OSMnx and a raster required by pylandstats is practically just a use of geocube's [`make_cube`](https://corteva.github.io/geocube/stable/getting_started.html), not? So momepy is not involved at any point. I would say that raster is out of scope for our project and if we had some usage for that, I would try to send a PR elsewhere, like to pylandstats directly.

cc @martibosch

**Comment by martibosch** [comment]
Hello @mohdalattar,

yes, `pylandstats` works with categorical land use/land cover **rasters**. Therefore, while it is perfectly possible to use it with OSM data, the latter needs to be rasterized first (like @martinfleis mentioned, this can be done with `make_cube` or also [with rasterio](https://rasterio.readthedocs.io/en/latest/topics/features.html#burning-shapes-into-a-raster).

Otherwise, there is a tool named [geograph](https://github.com/ai4er-cdt/geograph) (I have not used it myself) which can compute landscape metrics from vector data.

I hope this helps. Best,
Martí
